File: SYP.Web/Modules/ServerTypes/Order/OrderDocumentColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderDocumentRow } from "./OrderDocumentRow";

export interface OrderDocumentColumns {
    Id: Column<OrderDocumentRow>;
    OrderNumber: Column<OrderDocumentRow>;
    DocumentType: Column<OrderDocumentRow>;
    FileName: Column<OrderDocumentRow>;
    FilePath: Column<OrderDocumentRow>;
    FileSize: Column<OrderDocumentRow>;
    MimeType: Column<OrderDocumentRow>;
    UploadedByUserUsername: Column<OrderDocumentRow>;
    UploadDate: Column<OrderDocumentRow>;
    IsActive: Column<OrderDocumentRow>;
    Notes: Column<OrderDocumentRow>;
}

export class OrderDocumentColumns extends ColumnsBase<OrderDocumentRow> {
    static readonly columnsKey = 'Order.OrderDocument';
    static readonly Fields = fieldsProxy<OrderDocumentColumns>();
}

File: SYP.Web/Modules/Order/OrderDocument/OrderDocumentColumns.cs
namespace SYP.Order.Columns;

[ColumnsScript("Order.OrderDocument")]
[BasedOnRow(typeof(OrderDocumentRow), CheckNames = true)]
public class OrderDocumentColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    public string OrderNumber { get; set; }
    public int DocumentType { get; set; }
    [EditLink]
    public string FileName { get; set; }
    public string FilePath { get; set; }
    public long FileSize { get; set; }
    public string MimeType { get; set; }
    public string UploadedByUserUsername { get; set; }
    public DateTime UploadDate { get; set; }
    public bool IsActive { get; set; }
    public string Notes { get; set; }
}